src/compute-engine/global-types.ts

```diff
@@ -1048,12 +1048,24 @@ export interface BoxedExpression {
    *
    * Infer the type of this expression.
    *
-   * If the type of this expression is already known, return `false`.
+   * Effective only for overall BoxedExpression types which are *non-constant* and therefore for
```

Comment:
Do please verify, or amend.

Reply:
`for functions, narrows the *(return) type*`, may be incorrect

Reply:
I believe that's correct